### DISCUSSION TITLE
Network access in Web

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [fzyzcjy/flutter_rust_bridge#1484] [Q&A] [answered]
Hello,

I'm exploring the Flutter Rust Bridge for web development and have concerns about network access limitations in WASM. I'm curious about:

1. How does the lack of network access in WASM impact projects using Flutter Rust Bridge?
2. Are there any known workarounds or solutions to this issue?

Looking forward to your insights.

Thanks!

**Comment by fzyzcjy** [comment, marked as answer]
Good question!

Firstly, there seems to be some Rust crates to wrap the native js's `fetch`, etc https://stackoverflow.com/questions/66397775/how-do-i-make-an-http-request-within-a-wasm-bindgen-function-in-rust

Secondly, FRB v2 introduces "Rust call Dart" feature https://cjycode.com/flutter_rust_bridge/guides/direction/rust-call-dart. So you can write a Dart function that does network request (e.g. `Future<String> myDartFunctionThatReqeusts(String url) { ... }`, and call it from Rust.

**Reply by vincent-herlemont** [reply]
@fzyzcjy Thanks :pray:

**Reply by fzyzcjy** [reply]
You are welcome!
